letsencrypt/scripts/main.py

```diff
@@ -122,91 +135,57 @@ def main():  # pylint: disable=too-many-branches
     if not args.eula:
         display_eula()
 
-    # Make sure we actually get an installer that is functioning properly
-    # before we begin to try to use it.
-    try:
-        installer = client.determine_installer(config)
-    except errors.LetsEncryptMisconfigurationError as err:
-        logging.fatal("Please fix your configuration before proceeding.  "
-                      "The Installer exited with the following message: "
-                      "%s", err)
-        sys.exit(1)
+    # list of (Description, Known Authenticator classes, init arguments)
+    all_auths = [
+        ("Apache Web Server", configurator.ApacheConfigurator, config),
+        ("Standalone Authenticator", standalone.StandaloneAuthenticator),
+    ]
+    auth = client.determine_authenticator(all_auths)
+    if auth is None:
+        logging.critical("Unable to find a way to authenticate the server.")
+        sys.exit(4)
```

Comment:
No good reason :) I don't think documenting is a bad idea... but this exit probably won't exist after a few more PRs.  I have been wrestling with the determine_authenticator function (as you have seen) and it probably makes more sense to throw an error if it isn't available.  Thus don't ask for us to find an authenticator unless you really need one.

When --install-only is defined this code snippet should probably be skipped entirely.  Since I am going to have to rework the Authenticators/Installers anyway in order to make the secondary __init__ function.  I will modify determine_authenticator to simply throw an error right now if None are available.